types/recommendationrequest.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2019 The MayaData Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package types
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CStorClusterRecommendationRequest is a kubernetes custom
+// resource that represents configuration input or requirement
+// to create or scale one CStor pool.
+type CStorClusterRecommendationRequest struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec CStorClusterRecommendationRequestSpec `json:"spec"`
+}
+
+// CStorClusterRecommendationRequestSpec contains input or requirement
+// to create or scale one CStorpoolcluster.
+type CStorClusterRecommendationRequestSpec struct {
+	// Name represents name of the cspc.
+	Name string `json:"name"`
+	// Namespace represents namespace of the cspc.
+	Namespace string `json:"namespace"`
+	// PoolCapacity represents requested capacity for one pool
+	PoolCapacity resource.Quantity `json:"capacity"`
```

Comment:
Is PoolCapacity & DeviceCount optional to each other? In other words, if PoolCapacity is mentioned then DeviceCount becomes optional? How do we ensure we accepts a 100GiB disk or two 50GiB disks for a PoolCapacity of 100GiB?

Should PoolCapacity be a generated value instead? i.e. move it out from Spec to Status?
Why is PoolCapacity preferred over DiskCapacity?

Reply:
We are not taking device count as input. should we add that to get better recommendations?
If the user is not providing this than we can say by default we will pick the larger capacity device.
IMO we can add this field good point.
cc @avishnu @AmitKumarDas.